p2p/src/main/java/bisq/network/p2p/P2PService.java

```diff
@@ -337,11 +340,13 @@ public void onPreliminaryDataReceived() {
 
     @Override
     public void onUpdatedDataReceived() {
-        isBootstrapped = true;
-        maybeProcessAllMailboxEntries();
-        p2pServiceListeners.stream().forEach(P2PServiceListener::onUpdatedDataReceived);
-        if(!isBootstrapped)
-            p2PDataStorage.onBootstrapComplete();
+        if (!isBootstrapped() || Capabilities.app.containsAll(Capability.SEED_NODE)) {
+            isBootstrapped = true;
+            maybeProcessAllMailboxEntries();
+            p2pServiceListeners.stream().forEach(P2PServiceListener::onUpdatedDataReceived);
+            if (!isBootstrapped)
+                p2PDataStorage.onBootstrapComplete();
+        }
```

Comment:
In your new code it is not executed is isBootStapped=true and not a seed node. Without having looked to a wider context that seems wrong or at lease a behaviour change.

Reply:
Ah ignore previous comment... See in the first commit the original code...

Reply:
Maybe worth to squash all commits, makes review easier

Reply:
Ahm I am not a fan of squashing commits, because:
- if a reviewer wants to see individual commits (because there is far less a change than all of them), he can just click them
- if a reviewer wants to see all changes, he can review in the [Files]-Tab...

as for your comment, these `if`s are strange. Onto it.

Reply:
fixed by doing
```
        if (!isBootstrapped() || Capabilities.app.containsAll(Capability.SEED_NODE)) {
            maybeProcessAllMailboxEntries();
            p2pServiceListeners.stream().forEach(P2PServiceListener::onUpdatedDataReceived);
        }

        if (!isBootstrapped()) {
            isBootstrapped = true;
            p2PDataStorage.onBootstrapComplete();
        }
```